File: frontend/src/api/rules.ts
import { getData } from './http'

export interface RuleRow {
  antecedents: string
  consequents: string
  support: number
  confidence: number
  lift: number
}

export interface RulesData {
  method?: string
  n_rules?: number
  min_lift?: number
  disclaimer: string
  rules: RuleRow[]
}

export function fetchRules(minLift = 1.0, limit = 50) {
  return getData<RulesData>('/rules', { min_lift: minLift, limit })
}

File: src/digital_marketing/schemas/rules.py
"""关联规则 DTO。"""

from __future__ import annotations

from typing import Any

from pydantic import BaseModel, Field


class RulesData(BaseModel):
    method: str | None = None
    n_rules: int | None = None
    min_lift: float | None = None
    disclaimer: str = "关联规则表达相关而非因果"
    rules: list[dict[str, Any]] = Field(default_factory=list)
